internal/workflow_testsuite.go

```diff
@@ -388,19 +388,26 @@ func (e *TestWorkflowEnvironment) OnWorkflow(workflow interface{}, args ...inter
 	var call *mock.Call
 	switch fType.Kind() {
 	case reflect.Func:
-		fnType := reflect.TypeOf(workflow)
-		if err := validateFnFormat(fnType, true); err != nil {
+		if err := validateFnFormat(fType, true); err != nil {
 			panic(err)
 		}
 		fnName, _ := getWorkflowFunctionName(e.impl.registry, workflow)
 		if alias, ok := e.impl.registry.getWorkflowAlias(fnName); ok {
 			fnName = alias
 		}
+		e.impl.registry.RegisterWorkflowWithOptions(workflow, RegisterWorkflowOptions{DisableAlreadyRegisteredCheck: true})
 		call = e.mock.On(fnName, args...)
 	case reflect.String:
-		call = e.mock.On(workflow.(string), args...)
+		workflowName := workflow.(string)
+		_, ok := e.impl.registry.getWorkflowFn(workflowName)
+		if !ok {
+			registered := strings.Join(e.impl.registry.getRegisteredWorkflowTypes(), ", ")
+			panic(fmt.Sprintf("Workflow \""+workflowName+"\" is not registered with the TestWorkflowEnvironment, "+
```

Comment:
My concern here is people might have a test case that sets `OnWorkflow` as a string before they register it. And now that's gonna panic because they did it in that order. I think for compatibility reasons we can't panic here. We have to leave today's behavior as it was if possible.

Reply:
👍  agreed, didn't think about, it let me fix

Reply:
Nah, I think you can just silently let it behave like today

Reply:
Fixed with https://github.com/temporalio/sdk-go/pull/1128/commits/a820a9c85268b6d660edd2f18bd6a66baf4e692a